Add geopmhash regression fuzz test

avilcheslopez opened the pull request:
Move fuzz test to its own directory, create separate regression test.

- Fixes #3060

avilcheslopez committed:
Add geopmhash regression fuzz test

Move fuzz test to its own directory, create separate regression
test.

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Renamed fuzz_tests directory to fuzz_test

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Fix fuzz_test paths in makefiles

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Move geopmhash fuzz test corpus out of repo

- Move corpus to geopm.github.io repo, download as part of
  build
- Add apache2-llvm license into TPP file, other files for
  license checks

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Update MANIFEST.EXEMPT file to include new license files

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Fix test-dist, add fuzz test run script

- Add new apache2-llvm license files to MANIFEST.EXEMPT
- Add helper script to run fuzz regression tests

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo commented on service/fuzz_test/run_fuzz_tests.sh:
```diff
@@ -0,0 +1,18 @@
+
+test_dir=$(dirname $0)
+
+for t in ${test_dir}/*_reg_test; do
+    ${t} ${t/reg_test/corpus}/* >& ${t}.log
+    result=$?
```
```suggestion
for t in ${test_dir}/*_reg_test; do
    ${t} ${t}/reg_test/corpus/* >& ${t}.log
    result=$?
```
?

cmcantalupo reviewed (commented):
LGTM except maybe there is an issue with the bash script to run the fuzz tests.

avilcheslopez committed:
Add run_fuzz_tests.sh to dist

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo commented on service/fuzz_test/Makefile.mk:
```diff
@@ -0,0 +1,40 @@
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+fuzz_test_dir = fuzz_test
```
This parameter is used on lines 7-9 but not lines 20 - 25.  I'd be okay omitting the "fuzz_test_dir" variable, or using it throughout in place of "fuzz_test"

cmcantalupo commented on service/Makefile.am:
```diff
@@ -79,6 +79,7 @@ EXTRA_DIST = .gitignore \
              debian/rules \
              dox/blurb.md \
              dox/Doxyfile.in \
+             fuzz_test/run_fuzz_tests.sh \
```
since we have a fuzz_test/Makefile.mk maybe add

```
EXTRA_DIST += fuzz_test/run_fuzz_test.sh
```

to fuzz_test/Makefile.mk?

cmcantalupo reviewed (commented):
more review, more comments ....

avilcheslopez committed:
Delegate fuzz dist configuration to fuzz_test/Makefile.mk

Manage conditional inclusion/configuration of fuzz testing files and
targets in fuzz_test/Makefile.mk (as opposed to higher level Makefile).

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>
